sdk/go/common/workspace/plugins.go

```diff
@@ -1173,9 +1173,28 @@ func GetPluginInfo(kind PluginKind, name string, version *semver.Version) (*Plug
 //  * if found in the pulumi dir's installed plugins, a PluginInfo and path to the executable
 //  * an error in all other cases.
 func getPluginInfoOrPath(
-	kind PluginKind, name string, version *semver.Version, skipMetadata bool) (*PluginInfo, string, error) {
+	kind PluginKind, name string, version *semver.Version, skipMetadata bool, projectPlugins map[string]*PluginInfo) (*PluginInfo, string, error) {
 	var filename string
 
+	if projectPlugins != nil {
+		plugin, ok := projectPlugins[name]
+		if ok {
+			if plugin.Name != name {
+				return nil, "", fmt.Errorf("plugin name mismatch: %s != %s", plugin.Name, name)
+			}
+			if plugin.Kind != kind {
+				return nil, "", fmt.Errorf("plugin kind mismatch: %s != %s", plugin.Kind, kind)
+			}
+			//Comparing pointers? Seems wrong, but I get problems when I try to dereference
```

Comment:
Yeh either version can be nil so you need to check for that first. If you look down around line 1280  there's some "if nil" checks before calling Version.GTE()